### ISSUE TITLE
Trimming the tree

### ISSUE THREAD

**Issue by eli-s-goldberg** [PandaStabber/Goldberg_et_al_2016#48]
Decision trees are often overfit in scikit learn because trimming isn't implemented. One of the reasons it isn't, is because more powerful and robust predictive ensemble methods exist (random forest/ gradient boosting). However, it's actually really useful for helping to understand the data. There are a few implementations of trimming in other ML packages/languages (I think they are more common in R). This would take some effort, but it might be a more interesting problem to tackle.

**Comment by eli-s-goldberg** [comment]
Also, it's pruning... Not trimming. I was just trying to be clever. Maybe we should see if another's implementation works? https://github.com/scikit-learn/scikit-learn/pull/941

**Comment by eli-s-goldberg** [comment]
https://triangleinequality.wordpress.com/2013/09/01/decision-trees-part-3-pruning-your-tree/
